doc/cascadia/Json-Utility-API.md

```diff
@@ -0,0 +1,163 @@
+# New Json Utility API
+
+## Raw value conversion (GetValue)
+
+`GetValue` is a convenience helper that will either read a value into existing storage (type-deduced) or
+return a JSON value coerced into the specified type.
+
+When reading into existing storage, it returns a boolean indicating whether that storage was modified.
+
+If the JSON value cannot be converted to the specified type, an exception will be generated.
+
+```c++
+std::string one;
+std::optional<std::string> two;
+
+JsonUtils::GetValue(json, one);
+// one is populated or unchanged.
+
+JsonUtils::GetValue(json, two);
+// two is populated, nullopt or unchanged
+
+auto three = JsonUtils::GetValue<std::string>(json);
+// three is populated or zero-initialized
+
+auto four = JsonUtils::GetValue<std::optional<std::string>>(json);
+// four is populated or nullopt
+```
+
+## Key lookup (GetValueForKey)
+
+`GetValueForKey` follows the same rules as `GetValue`, but takes an additional key.
+It is assumed that the JSON value passed to GetValueForKey is of `object` type.
+
+```c++
+std::string one;
+std::optional<std::string> two;
+
+JsonUtils::GetValueForKey(json, "firstKey", one);
+// one is populated or unchanged.
+
+JsonUtils::GetValueForKey(json, "secondKey", two);
+// two is populated, nullopt or unchanged
+
+auto three = JsonUtils::GetValueForKey<std::string>(json, "thirdKey");
+// three is populated or zero-initialized
+
+auto four = JsonUtils::GetValueForKey<std::optional<std::string>>(json, "fourthKey");
+// four is populated or nullopt
+```
+
+## Converting User-Defined Types
+
+All conversions are done using specializations of `JsonUtils::ConversionTrait<T>`.
+To implement a converter for a user-defined type, you must implement a specialization of `JsonUtils::ConversionTrait<T>`.
+
+Every specialization over `T` must implement `static T FromJson(const Json::Value&)` and
+`static bool CanConvert(const Json::Value&)`.
+
+```c++
+template<>
+struct ConversionTrait<MyCustomType>
+{
+    // This trait converts a string of the format "[0-9]" to a value of type MyCustomType.
+
+    static MyCustomType FromJson(const Json::Value& json)
+    {
+        return MyCustomType{ json.asString()[0] - '0' };
+    }
+
+    static bool CanConvert(const Json::Value& json)
+    {
+        return json.isString();
+    }
+};
+```
+
+For your "convenience" (;P), if you need to provide name-value enum mapping,
+there's the `EnumMapper<>` base template. It is somewhat verbose.
+
+```c++
+template<>
+struct JsonUtils::ConversionTrait<CursorStyle> : public JsonUtils::EnumMapper<CursorStyle, JsonUtils::ConversionTrait<CursorStyle>>

```

Comment:
Hey this seems like a perfect place for a macro to just fill this in for you, because yea, this is kinda verbose

Reply:
(I'm thinking like the other `cppwinrt_utils`)

Reply:
Yeah, I like this. It's clever.